legion/engine/rendering/data/particle_system_cache.cpp

```diff
@@ -0,0 +1,26 @@
+#include <rendering/data/particle_system_cache.hpp>
+#include <rendering/data/particle_system_base.hpp>
+
+namespace legion::rendering
+{
+    std::unordered_map<id_type, std::unique_ptr<const ParticleSystemBase>> ParticleSystemCache::m_cache;
+    async::readonly_rw_spinlock ParticleSystemCache::m_particleSystemLock;
+
+    ParticleSystemHandle ParticleSystemCache::createParticleSystemImpl(std::string name, ParticleSystemBase* ptr)
+    {
+        std::unique_ptr<ParticleSystemBase> particleSystem(ptr);
+        id_type id = nameHash(name);
+
+        //acquire lock
```

Comment:
we should probably check if the particle system doesn't exist already before we create a new one